### PR TITLE
[GLUTEN-2919][VL]Fix incorrect hive scan fallback or offload for velox

### PR DESCRIPTION
**Pull request by yma11**
## What changes were proposed in this pull request?
Due to some code change for `HiveTableScanExecTransformer`, like [PR](https://github.com/oap-project/gluten/pull/2773) and [PR](https://github.com/oap-project/gluten/pull/2558), etc, scan offloading behavior becomes incorrect, such as ORC format scan can't be offloaded anymore. The root cause is that some  ClickHouse format/datatype check logic is added in this Op which is supposed to add in CKBackend API instead. This PR intends to fix it.

## How was this patch tested?
UT

### CODE REVIEW

File: gluten-ut/spark32/src/test/scala/org/apache/spark/sql/hive/execution/GlutenHiveSQLQuerySuite.scala [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+          " USING hive OPTIONS(fileFormat 'orc')")
+      sql("INSERT INTO test_orc VALUES('test_1', 'red')");
+      val df = spark.sql("select * from test_orc")
+      checkAnswer(df, Seq(Row("test_1", "red")))
+      checkOperatorMatch[HiveTableScanExecTransformer](df)
+    }
+    spark.sessionState.catalog.dropTable(
+      TableIdentifier("test_orc"),
+      ignoreIfNotExists = true,
+      purge = false)
```
Comment (winningsix) [inline gluten-ut/spark32/src/test/scala/org/apache/spark/sql/hive/execution/GlutenHiveSQLQuerySuite.scala:165]: hint: ```use purge = true``` avoiding using Trash directory?

File: gluten-ut/spark32/src/test/scala/org/apache/spark/sql/hive/execution/GlutenHiveSQLQuerySuite.scala [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+  }
+
+  test("hive orc scan") {
```
Comment (winningsix) [inline gluten-ut/spark32/src/test/scala/org/apache/spark/sql/hive/execution/GlutenHiveSQLQuerySuite.scala:151]: hmm, duplicated test cases?

File: gluten-core/src/main/scala/org/apache/spark/sql/hive/HiveTableScanExecTransformer.scala [hunk 3]
```diff
@@ -152,6 +156,12 @@ class HiveTableScanExecTransformer(
         } else {
           ValidationResult.notOk("does not support complex type")
         }
+      case ReadFileFormat.OrcReadFormat =>
+        if (!hasComplexType) {
+          ValidationResult.ok
+        } else {
+          ValidationResult.notOk("does not support complex type")
```
[the remark this answers is not in the document]
Comment (yma11) [reply gluten-core/src/main/scala/org/apache/spark/sql/hive/HiveTableScanExecTransformer.scala]: Updated.

File: gluten-ut/spark32/src/test/scala/org/apache/spark/sql/hive/execution/GlutenHiveSQLQuerySuite.scala [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+      .set(UI_ENABLED, false)
+      .set(config.UNSAFE_EXCEPTION_ON_MEMORY_LEAK, true)
+      // Hive changed the default of hive.metastore.disallow.incompatible.col.type.changes
+      // from false to true. For details, see the JIRA HIVE-12320 and HIVE-17764.
+      .set("spark.hadoop.hive.metastore.disallow.incompatible.col.type.changes", "false")
+      // Disable ConvertToLocalRelation for better test coverage. Test cases built on
+      // LocalRelation will exercise the optimization rules better by disabling it as
+      // this rule may potentially block testing of other optimization rules such as
+      // ConstantPropagation etc.
+      .set(SQLConf.OPTIMIZER_EXCLUDED_RULES.key, ConvertToLocalRelation.ruleName)
```
[the remark this answers is not in the document]
Comment (yma11) [reply gluten-ut/spark32/src/test/scala/org/apache/spark/sql/hive/execution/GlutenHiveSQLQuerySuite.scala:69]: `VeloxParquetWriteForHiveSuite `  is in a different module `backends-velox` and these test utils are not passed as dependency in these modules.  But we can add a `GlutenHiveSQLQuerySuiteTrait `  for Hive related test cases later.

File: gluten-ut/spark32/src/test/scala/org/apache/spark/sql/hive/execution/GlutenHiveSQLQuerySuite.scala (hunk 1 above)
Comment (yma11) [reply gluten-ut/spark32/src/test/scala/org/apache/spark/sql/hive/execution/GlutenHiveSQLQuerySuite.scala:165]: This is following vanilla spark UTs like `ParquetQuerySuite`/`OrcQuerySuite`, etc.

File: gluten-ut/spark32/src/test/scala/org/apache/spark/sql/hive/execution/GlutenHiveSQLQuerySuite.scala (hunk 2 above)
Comment (yma11) [reply gluten-ut/spark32/src/test/scala/org/apache/spark/sql/hive/execution/GlutenHiveSQLQuerySuite.scala:135]: Removed.

**Comment by winningsix** [comment]
Thanks for updates. LGTM.

**Review by FelixYBW — APPROVED** [review]
verified from customer queries

**Comment by yma11** [comment]
> verified from customer queries

Hi binwei, this PR failed some CK UTs as changed some code in common path. I need do a update today. Please pending on merge.

File: gluten-core/src/main/scala/org/apache/spark/sql/hive/HiveTableScanExecTransformer.scala [hunk 5]
```diff
@@ -122,40 +123,13 @@ class HiveTableScanExecTransformer(
...
-    fileFormat match {
-      case ReadFileFormat.JsonReadFormat => ValidationResult.ok
-      case ReadFileFormat.TextReadFormat =>
-        if (!hasComplexType) {
-          ValidationResult.ok
-        } else {
-          ValidationResult.notOk("does not support complex type")
-        }
-      case _ => ValidationResult.notOk("Unknown file format")
-    }
```
Comment (yma11) [inline gluten-core/src/main/scala/org/apache/spark/sql/hive/HiveTableScanExecTransformer.scala:156]: above check is specific for ClickHouse backend so moved to corresponding backend API instead.

**Comment by yma11** [comment]
@zhouyuan please help take a look. Thanks.

File: gluten-core/src/main/scala/org/apache/spark/sql/hive/HiveTableScanExecTransformer.scala (hunk 5 above)
Comment (zhouyuan) [reply gluten-core/src/main/scala/org/apache/spark/sql/hive/HiveTableScanExecTransformer.scala:156]: CC: @zzcclp this piece of code moved to CK backend

File: backends-clickhouse/src/main/scala/io/glutenproject/backendsapi/clickhouse/CHBackend.scala [hunk 6]
```diff
@@ -98,11 +98,28 @@ object CHBackendSettings extends BackendSettingsApi with Logging {
...
+    def hasComplexType: Boolean = {
+      // Collect unsupported types.
+      val unsupportedDataTypes = fields.map(_.dataType).collect {
+        case _: MapType => "MapType"
+        case _: StructType => "StructType"
+        case _: ArrayType => "ArrayType"
+      }
+      for (unsupportedDataType <- unsupportedDataTypes) {
+        // scalastyle:off println
+        println(
```
Comment (zhouyuan) [inline backends-clickhouse/src/main/scala/io/glutenproject/backendsapi/clickhouse/CHBackend.scala:111]: is the `println ` for debug purpose?

**Review by zhouyuan — COMMENTED** [review]
thanks for improving on this!

**Comment by zzcclp** [comment]
@taiyang-li @lgbo-ustc please help to review thanks.

File: backends-clickhouse/src/main/scala/io/glutenproject/backendsapi/clickhouse/CHBackend.scala (hunk 6 above)
Comment (yma11) [reply backends-clickhouse/src/main/scala/io/glutenproject/backendsapi/clickhouse/CHBackend.scala:111]: Not really. This is necessary to print out the fallback reason directly as the info hard to pass back to FallbackReporter.

File: gluten-core/src/main/scala/org/apache/spark/sql/hive/HiveTableScanExecTransformer.scala (hunk 5 above)
Comment (lgbo-ustc) [reply gluten-core/src/main/scala/org/apache/spark/sql/hive/HiveTableScanExecTransformer.scala:156]: LGTM

File: backends-clickhouse/src/main/scala/io/glutenproject/backendsapi/clickhouse/CHBackend.scala (hunk 6 above)
Comment (lgbo-ustc) [reply backends-clickhouse/src/main/scala/io/glutenproject/backendsapi/clickhouse/CHBackend.scala:111]: can we use log to print the message?

**Comment by lgbo-ustc** [comment]
LGTM

**Review by zhouyuan — APPROVED** [review]
👍

**Pull request merged**